docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/EntityFrameworkEnumerable.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public class HousingData
+    {
+        // Entity Framework requires the object to have a primary key
+        public int HousingDataId { get; set; }
+        public float MedianHomeValue { get; set; }
+        public float CrimesPerCapita { get; set; }
+        public float PercentResidental { get; set; }
+        public float PercentNonRetail { get; set; }
+        public float CharlesRiver { get; set; }
+        public float NitricOxides { get; set; }
+        public float RoomsPerDwelling { get; set; }
+        public float PercentPre40s { get; set; }
+        public float EmploymentDistance { get; set; }
+        public float HighwayDistance { get; set; }
+        public float TaxRate { get; set; }
+        public float TeacherRatio { get; set; }
+    }
+
+    public partial class MLNetExampleContext : DbContext
+    {
+        public DbSet<HousingData> Housing { get; set; }
+
+        public MLNetExampleContext(DbContextOptions<MLNetExampleContext> options)
+            : base(options)
+        {
+        }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+                optionsBuilder.UseSqlServer("Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory;Trusted_Connection=True;ConnectRetryCount=0");
+            }
+        }
+    }
+
+    public static class EntityFrameworkEnumerable
+    {
+        private static IEnumerable<HousingData> housingData;
+
+        public static void Example()
+        {
+            // Download the housing data set and get the file name.
+            var housingFile = SamplesUtils.DatasetUtils.DownloadHousingRegressionDataset();
+
+            // Read data from file into an IEnumerable.
+            var data = File.ReadAllLines(housingFile)
+                .Skip(1) // Skip the header row
+                .Select(l => l.Split('\t'))
+                .Select(i => new HousingData
+                {
+                    MedianHomeValue = float.Parse(i[0]),
+                    CrimesPerCapita = float.Parse(i[1]),
+                    PercentResidental = float.Parse(i[2]),
+                    PercentNonRetail = float.Parse(i[3]),
+                    CharlesRiver = float.Parse(i[4]),
+                    NitricOxides = float.Parse(i[5]),
+                    RoomsPerDwelling = float.Parse(i[6]),
+                    PercentPre40s = float.Parse(i[7])
+                });
+
+            // Options to pass to the DbContext to use an in-memory database for
+            // Entity Framework and to give the database a name.
+            var options = new DbContextOptionsBuilder<MLNetExampleContext>()
```

Comment:
You may want to consider moving this inside the OnConfiguring method for simplicity, replacing the code currently there. On the other hand, I think an example that uses the in-memory provider (which BTW is recommended only for testing) misses an opportunity to show when you would actually want to pull the data from a real database using EF Core, which I believe @singlis's version of the sample addresses.